di-ipv-ticf-stub/lambdas/src/services/ticfService.ts

```diff
@@ -0,0 +1,120 @@
+    // preparing response
+    let ticfSigningKey: string;
+    try {
+      ticfSigningKey = await getSsmParameter(
+        process.env.TICF_PARAM_BASE_PATH + "signingKey"
+      );
+    } catch (error) {
+      throw new Error(`Error while retrieving TicF CRI VC signing key.`);
+    }
+    let timeoutVC: string | null | undefined = await getSsmParameter(
+      process.env.TICF_PARAM_BASE_PATH + "timeoutVC"
+    );
+    timeoutVC ??= "false";
+    let includeCIToVC: string | null | undefined = await getSsmParameter(
+      process.env.TICF_PARAM_BASE_PATH + "includeCIToVC"
+    );
+    includeCIToVC ??= "false";
+
+    let buildJwtParams: SignedJwtParams;
+    buildJwtParams = {
+      issuer: process.env.ISSUER,
+      customClaims: getCustomClaims(
+        JSON.parse(timeoutVC.toLowerCase()),
+        JSON.parse(includeCIToVC.toLowerCase()),
+        ticfRequest.sub
+      ),
+      privateSigningKey: ticfSigningKey!,
+    };
+
+    // preparing response
+    const returnJwt = await buildSignedJwt(buildJwtParams);
+    let responseBody: TicfResponse;
+    responseBody = {
+      sub: ticfRequest.sub,
+      govuk_signin_journey_id: ticfRequest.govuk_signin_journey_id,
+      vtr: ticfRequest.vtr,
+      vot: ticfRequest.vot,
+      vtm: ticfRequest.vtm,
+      "https://vocab.account.gov.uk/v1/credentialJWT": [returnJwt],
+    };
+    return responseBody;
+  }
+}
+
+function getCustomClaims(
+  timeOutVC: boolean,
+  includeCIToVC: boolean,
+  userId: string
+): JWTPayload {
+  if (timeOutVC) {
+    return {
+      sub: userId,
+      iss: process.env.ISSUER,
+      aud: "https://development-di-ipv-core-front.london.cloudapps.digital",
+      nbf: Date.now(),
+      vc: {
+        evidence: [
+          {
+            type: "RiskAssessment",
+          },
+        ],
+        type: ["VerifiableCredential", "RiskAssessmentCredential"],
+      },
+      jti: "urn:uuid:" + uuid(),
+    };
+  } else {
+    if (includeCIToVC) {
+      return {
+        sub: userId,
+        iss: process.env.ISSUER,
+        aud: "https://development-di-ipv-core-front.london.cloudapps.digital",
+        nbf: Date.now(),
+        vc: {
+          evidence: [
+            {
+              type: "RiskAssessment",
+              ci: ["V03"],
+              txn: uuid(),
+            },
+          ],
+          type: ["VerifiableCredential", "RiskAssessmentCredential"],
+        },
+        jti: "urn:uuid:" + uuid(),
+      };
+    } else {
+      return {
+        sub: userId,
+        iss: process.env.ISSUER,
+        aud: "https://development-di-ipv-core-front.london.cloudapps.digital",
+        nbf: Date.now(),
+        vc: {
+          evidence: [
+            {
+              type: "RiskAssessment",
+              txn: uuid(),
+            },
+          ],
+          type: ["VerifiableCredential", "RiskAssessmentCredential"],
+        },
+        jti: "urn:uuid:" + uuid(),
+      };
+    }
+  }
+}
```

Comment:
There's still repetition here, that I think we can avoid?
```
function getCustomClaims(
  timeOutVC: boolean,
  includeCIToVC: boolean,
  userId: string
): JWTPayload {
  return {
    sub: userId,
    iss: process.env.ISSUER,
    aud: "https://development-di-ipv-core-front.london.cloudapps.digital",
    nbf: Date.now(),
    vc: {
      evidence: [getEvidenceItem(timeOutVC, includeCIToVC)],
      type: ["VerifiableCredential", "RiskAssessmentCredential"],
    },
    jti: "urn:uuid:" + uuid(),
  };
}

function getEvidenceItem(
  timeOutVC: boolean,
  includeCIToVC: boolean
): TicfEvidenceItem {
  if (timeOutVC) {
    return { type: "RiskAssessment" };
  } else {
    if (includeCIToVC) {
      return {
        type: "RiskAssessment",
        ci: ["V03"],
        txn: uuid(),
      };
    } else {
      return {
        type: "RiskAssessment",
        txn: uuid(),
      };
    }
  }
}
```
This should also make it easy to update when the evidence block comes from the management API

Reply:
done